cfpb/design-system
# Automate generation of the side navigation

<img width="691" alt="Screen Shot 2019-08-09 at 1 27 32 PM" src="https://user-images.githubusercontent.com/1060248/62799087-e116ff00-baad-11e9-9506-582aabcb585e.png">

## Adds

- CMS select widget called [`secondary_section`](https://github.com/cfpb/design-system/pull/204/files#diff-608a17279ac4f612293ab44079a7645cR35) for choosing a page's secondary nav parent.
- Jekyll [generator plugin](https://github.com/cfpb/design-system/pull/204/files#diff-20113a3432ceb7134937632a56aef6de) to create filenames for pages from their 'section' and 'title' front matter fields.

## Removes

- `permalink` widget
- `navigation` collection and `navigation.yml` file
- Obsolete `collection_name` and `permalink` front matter entries from all page files.

## Testing

1. Go to a component page at https://test-things.github.io/design-system/
1. Notice the component's location in the side nav.
1. Edit the component and change the navigation section. Save the page.
1. Go to the PR you created at https://github.com/test-things/design-system/pulls
1. Wait for the Netlify bot to create a preview link for you. Visit your component page via that preview link and it should be in the new side nav section you selected.
